hawkular-inventory-api/src/main/java/org/hawkular/inventory/api/Inventory.java

```diff
@@ -101,6 +110,91 @@
  * @since 0.0.1
  */
 public interface Inventory extends AutoCloseable, Tenants.Container<Tenants.ReadWrite> {
+    static final class InventotyUtils {
+        private static final Map<String, Class<? extends Entity<?, ?>>>entityTypes;
+        static {
+            Map<String, Class<? extends Entity<?, ?>>> m = Arrays.asList(
+                    Tenant.class,
+                    Environment.class,
+                    Feed.class,
+                    ResourceType.class,
+                    MetricType.class,
+                    Resource.class,
+                    Metric.class,
+                    OperationType.class,
+                    MetadataPack.class)
+            .stream().collect(Collectors.toMap(Class::getName, Function.identity()));
+            entityTypes = Collections.unmodifiableMap(m);
+        }
+        static String shortenIfNeeded(CanonicalPath path) {
+            String id = path.toString();
+
+            if (id.length() > 250) {
+                return UUID.nameUUIDFromBytes(id.getBytes()).toString();
+            } else {
+                return id;
+            }
+        }
+    }
+    static String getStableId(CanonicalPath path) {
+        return InventotyUtils.shortenIfNeeded(path);
+    }
+    static String getStableId(AbstractElement<?, ?> element) {
+        return getStableId(element.getPath());
+    }
+    static CanonicalPath toCanonicalPath(String restPath) {
```

Comment:
None in particular, because it will disappear anyway. The new REST API will not use it.